src/main.rs

```diff
@@ -95,7 +100,29 @@ impl State {
     }
 }
 
-fn main() -> Result<()> {
+fn main() -> Result<(), Box<dyn Error>> {
+    let raw_args = RawArgs::from_args();
+    let mut cursor = raw_args.cursor();
+
+    // Parse the arguments
+    while let Some(arg) = raw_args.next_os(&mut cursor) {
+        match arg.to_str() {
+            Some("--help") | Some("-h") => {
+                print_help(env!("CARGO_PKG_VERSION"), env!("VERGEN_GIT_SHA"));
+		return Ok(());
+            }
+            Some("--version") | Some("-v") => {
```

Comment:
Usually `-V` would be used for version in case `-v` is introduced later for `verbose`.